android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -42,59 +56,99 @@ public void setDescription(String description) {
 
     @ReactMethod
     public void setMetadata(ReadableMap metadataMap) throws JSONException {
-        Radar.setMetadata(RNRadarUtils.jsonObjectForMap(metadataMap));
-    }
-
-    @ReactMethod
-    public void setPlacesProvider(String providerStr) {
-        Radar.setPlacesProvider(RNRadarUtils.placesProviderForString(providerStr));
+        Radar.setMetadata(RNRadarUtils.jsonForMap(metadataMap));
     }
 
     @ReactMethod
     public void getPermissionsStatus(Promise promise) {
-        promise.resolve(RNRadarUtils.stringForPermissionsStatus(
-            ActivityCompat.checkSelfPermission(getReactApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
-        ));
+        if (promise == null) {
+            return;
+        }
+
+        boolean foreground = ActivityCompat.checkSelfPermission(getReactApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED;
+        if (Build.VERSION.SDK_INT >= 29) {
+            if (foreground) {
+                boolean background = ActivityCompat.checkSelfPermission(getReactApplicationContext(), Manifest.permission.ACCESS_BACKGROUND_LOCATION) == PackageManager.PERMISSION_GRANTED;
+                promise.resolve(background ? "GRANTED_BACKGROUND" : "GRANTED_FOREGROUND");
+            } else {
+                promise.resolve("DENIED");
+            }
+        } else {
+            promise.resolve(foreground ? "GRANTED_BACKGROUND" : "DENIED");
+        }
     }
 
     @ReactMethod
     public void requestPermissions(boolean background) {
```

Comment:
do we need the `background` param here?

Reply:
Good catch, yes, we need to check this before requesting background permissions. Will add.